src/Database/LSMTree/Internal/Config.hs

```diff
@@ -158,13 +158,26 @@ instance NFData WriteBufferAlloc where
 -- NOTE: a __physical__ database entry is a key\/operation pair that exists in a
 -- file, i.e., a run. Multiple physical entries that have the same key
 -- constitute a __logical__ database entry.
+--
+-- There is a trade-off between bloom filter memory size, and the false
+-- positive rate. A higher false positive rate (FPR) leads to more unnecessary
+-- I\/O. As a guide, here are some points on the trade-off:
+--
+-- * FPR of 1e-2 requires approximately 9.9 bits per element
+-- * FPR of 1e-3 requires approximately 15.8 bits per element
+-- * FPR of 1e-4 requires approximately 22.6 bits per element
+--
+-- The policy can be specified either by fixing a FPR or by fixing the number
+-- of bits per entry.
+--
 data BloomFilterAlloc =
     -- | Allocate a fixed number of bits per physical entry in each bloom
     -- filter.
     AllocFixed
       !Word64 -- ^ Bits per physical entry.
```

Comment:
And this must be at least 1? (or 0?)

Reply:
At least 0.

Reply:
I've documented this, and at the same time done the TODO to switch it from Word to Double, since fractional bits per key is perfectly ok.